### ISSUE TITLE
[cmake] Can't use feature ldaps on Windows

### ISSUE THREAD

**Issue by JackBoosY** [curl/curl#6284]
Hi guys:

Recently, we received an issue about curl (https://github.com/microsoft/vcpkg/issues/14030), that reported that the feature `ldaps`  could not be enabled on Windows.
Since curl in vcpkg uses cmake to build, I checked the cmake configuration in curl and found something strange:
https://github.com/curl/curl/blob/abd846c374c5269809ce11740754bc6f00a0b193/CMakeLists.txt#L511-L520
Here we can use the system's ldaps on windows, but openldap is not disabled here, and:
https://github.com/curl/curl/blob/abd846c374c5269809ce11740754bc6f00a0b193/CMakeLists.txt#L1398-L1400
There is no judgment variable `USE_WIN32_LDAP` when judging whether to enable ldaps.

I think that's caused this issue.
But even if I use `USE_WIN32_LDAP` and explicitly enable ldaps in the configuration log, using curl -V does not explicitly enable ldaps.
Configure log:
```
-- Enabled protocols: DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS MQTT POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP
```
Result:
```
curl 7.73.0-DEV (Windows) libcurl/7.73.0-DEV Schannel zlib/1.2.11
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher http https imap imaps ldap mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IPv6 Kerberos Largefile NTLM SPNEGO SSL SSPI TrackMemory libz
```

### I expected the following
ldaps should be enabled on Windows.

### curl/libcurl version
7.73.0

[curl -V output]
```
curl 7.73.0-DEV (Windows) libcurl/7.73.0-DEV Schannel zlib/1.2.11
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher http https imap imaps ldap mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IPv6 Kerberos Largefile NTLM SPNEGO SSL SSPI TrackMemory libz
```

### operating system
Windows 10, Visual Studio 2017, cmake 3.18.2

**Comment by bagder** [comment]
It seems nobody is going to work on this. I'll add it to `KNOWN_BUGS`.

**Comment by vszakats** [comment]
Proposing a fix for this in PR #10674.
